src/components/controllers/slotted-controller.ts

```diff
@@ -19,19 +23,40 @@ export class SlottedController implements ReactiveController {
   }
 
   onSlotChange() {
-    const elementName = this.hostEl.tagName;
-    const myElement: HTMLElement | null = this.hostEl.parentElement
-      ? this.hostEl.parentElement.querySelector(elementName)
-      : null;
-    if (myElement && myElement.shadowRoot) {
-      const slot: HTMLElement | null = myElement.shadowRoot.querySelector(
-        'slot'
-      )
-        ? myElement.shadowRoot.querySelector('slot')
-        : null;
-      if (slot) {
-        slot.append(...myElement.children);
-      }
+    // Search for all slots provided by the Web Component.
+    const slotsArray = this.hostEl.renderRoot.querySelectorAll('slot');
+
+    // If any slots found
+    if (slotsArray.length > 0) {
```

Comment:
Why check the length and then run a `foreach`? The foreach will do nothing if you have no items, so I don't think you need to check the length first.